src/vmm/src/vstate/vcpu/mod.rs

```diff
@@ -74,19 +77,51 @@ pub enum Error {
 pub type Result<T> = result::Result<T, Error>;
 
 /// Encapsulates configuration parameters for the guest vCPUS.
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug)]
 pub struct VcpuConfig {
     /// Number of guest VCPUs.
     pub vcpu_count: u8,
     /// Enable simultaneous multithreading in the CPUID configuration.
     pub smt: bool,
-    /// Hard-coded template to use.
-    pub static_cpu_template: CpuFeaturesTemplate,
-    /// Custom configuration for vCPU,
+    /// Configuration for vCPU
+    /// For x86_64 we need cpu configuration in any case
     #[cfg(target_arch = "x86_64")]
-    pub custom_cpu_config: Option<X86_64CpuConfiguration>,
+    pub custom_cpu_config: X86_64CpuConfiguration,
+    /// Custom configuration for vCPU,
     #[cfg(target_arch = "aarch64")]
-    pub custom_cpu_config: Option<Aarch64CpuConfiguration>,
+    pub custom_cpu_config: Aarch64CpuConfiguration,
+}
+
+impl VcpuConfig {
+    pub fn new(
+        vm_resources: &crate::resources::VmResources,
+        _vcpus: &[Vcpu],
+        _vmm: &crate::Vmm,
```

Comment:
Instantiating `VcpuConfig` for different architectures where some parameters are unused will be quite confusing for future maintainers. I really think we need to avoid this where we can.